File: src/main/kotlin/de/thm/mni/ii/classroom/event/ConferenceEvent.kt
package de.thm.mni.ii.classroom.event

import com.fasterxml.jackson.annotation.JsonIgnore
import de.thm.mni.ii.classroom.model.classroom.ConferenceInfo

data class ConferenceEvent(
    val conferenceInfo: ConferenceInfo,
    val conferenceAction: ConferenceAction
) : ClassroomEvent(ConferenceEvent::class.simpleName!!) {
    @JsonIgnore override fun getClassroomId(): String = this.conferenceInfo.classroomId
}

enum class ConferenceAction {
    CREATE,
    CLOSE,
    VISIBILITY,
    USER_CHANGE
}

File: web-gui/src/app/rsocket/event/ConferenceEvent.ts
import {ConferenceInfo} from "../../model/ConferenceInfo";
import {ClassroomEvent} from "./ClassroomEvent";

export enum ConferenceAction {
    CREATE = "CREATE",
    CLOSE = "CLOSE",
    PLENARY = "PLENARY",
    VISIBILITY = "VISIBILITY",
    USER_CHANGE = "USER_CHANGE"
}

export class ConferenceEvent extends ClassroomEvent {
    conferenceInfo: ConferenceInfo | undefined
    conferenceAction: ConferenceAction | undefined

    constructor() {
        super("ConferenceEvent");
    }
}
